### ISSUE TITLE
plantuml and image preview error 

### ISSUE THREAD

**Issue by uniqueck** [asciidoctor/asciidoctor-intellij-plugin#345]
First describe the behavior you see in the IDE or plugin (either a problem or a bug). 
if i have an directory like this:
```
src
│   └── docs
│       ├── images
│       │   └── mindmap.png
│       ├── index.adoc
│       ├── plantuml
│       │   └── mindmap.puml
│       └── subdirectory
│           ├── index.adoc
│           └── subsub
│               └── index.adoc
```

and the following content of index.adoc

```
ifndef::plantUMLDir[:plantUMLDir: .plantuml/]
ifndef::icons[:icons: font]
// ifndef::imagesdir[:imagesdir: ../../images]
// uncomment the ifndef above and the image should appear, but the plantuml has an error


==== Level 3 Heading
.generated plantuml diagram
[plantuml, "{plantUMLDir}mindmap3", png, align=center]
----
include::../../plantuml/mindmap.puml[]
----
.static image
image::mindmap.png[]
```
then i don't see in the preview editor the static image.

![MissingImageInPreview](https://user-images.githubusercontent.com/12494167/66688276-3a560700-ec86-11e9-8631-997b98d61625.png)

If i uncomment the following line `ifndef::imagesdir[:imagesdir: ../../images]` then i see the image, but an error for the plantuml diagram.

![ErrorOnPlantUmlDiagramInPreview](https://user-images.githubusercontent.com/12494167/66688277-3a560700-ec86-11e9-9384-cf447f74ee21.png)

If i call maven clean generate-resources then the generated files are fine.

[index.pdf](https://github.com/asciidoctor/asciidoctor-intellij-plugin/files/3719799/index.pdf)

One subdirectory above all is working as expected.

Then describe the behavior you expect or want to see in the IDE or plugin.

The expected behavior is that in all directory levels the preview editor is working as in the first or second directory level. 

[Repository with bug](https://github.com/uniqueck/asciidoctor-plantuml-examples/tree/bug/PlantUmlAndImages)

AsciiDoc Plugin Version: 0.30.7

IDE Name and Version: 
IntelliJ IDEA 2019.2.3 (Community Edition)
Build #IC-192.6817.14, built on September 24, 2019
Runtime version: 11.0.4+10-b304.69 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.15.0-65-generic
GC: ParNew, ConcurrentMarkSweep
Memory: 1923M
Cores: 8
Registry: ide.balloon.shadow.size=0
Non-Bundled Plugins: Lombook Plugin, MoreUnit, PlantUML integration, com.alayouni.ansiHighlight, com.euphoricity.gitignore, com.intellij.ideolog, de.docksnet.puml.syntaxchecker, gherkin, cucumber-java, lermitage.intellij.extra.icons, mobi.hsz.idea.gitignore, org.asciidoctor.intellij.asciidoc, org.intellij.RegexpTester, cucumber-groovy, com.cybrosis.catdea, BashSupport, org.sonarlint.idea

**Comment by ahus1** [comment]
Thanks for providing an example repository. Unfortunately your setup worked for me, I never saw the plantuml error in the preview. 

I was using Windows with the 0.30.9 version of the plugin. I'm using the JavaFX preview, looking at your screen shots it seems that you are using the JavaFX preview as well. 

Please give it a try with the latest version 0.30.9; this version also introduces auto-safe when switching to an Asciidoc document to keep the preview in sync with all files in the IDE.

One change I've made in the code: each plantuml diagram should have its own file name; two diagrams had the same file name in your example. But that shouldn't have caused the problem.

I've also migrated the setup to .asciidoctorconfig files, [see this plugin's wiki for some docs how this works](https://github.com/asciidoctor/asciidoctor-intellij-plugin/wiki/Support-project-specific-configurations). This keeps your sources clean, and you need only one .asciidoctorconfig that per directory, sometime even less than that. See https://github.com/uniqueck/asciidoctor-plantuml-examples/pull/1 for the change.

If the issue still occurs with the latest version 0.30.9, I'll boot up a linux system here and give it a try. What Linux distro do you use?

**Comment by uniqueck** [comment]
Hi Alexander,

thanks for your invest. I check it with the latest version and give it a try with the .asciidoctorconfig file feature. I use a Ubuntu distribution. 

Thank you for this great asciidoc plugin. It provides a great integration in our development environment.

**Comment by uniqueck** [comment]
Hi,
I tried the latest version 0.30.9 but the same behavior.
My current distro is Ubuntu 18.04.3 LTS.

**Comment by ahus1** [comment]
Over the next weeks I'll be short on time to dig into this. I label it "hacktoberfest" for anyone who wants to tackle and debug this.

**Comment by ahus1** [comment]
I was able to reproduce this on Ubuntu 18.04.3 LTS. The plugin sets outdir to keep all temporary generated files outside of the project directory. Unfortunately this conflicts with an imagesdir that traverses to a parent folder: This could lead to a situation where this traverses to a folder where the current user doesn't have write permissions any more, therefore the error message with the missing permissions regarding the /images folder. 

After setting another attribute imagesoutdir all temporary contents are now confined to the temporary folder. After stripping the imagesdir from the resulting HTML output I came up with this solution that works on Ubuntu 18.04.3 LTS (as well as on Windows) with the example you provided.

Pre-release 0.30.25 is now available that should fix this issue.

The pre-release of the plugin is available from [GitHub releases](https://github.com/asciidoctor/asciidoctor-intellij-plugin/releases) and the [IntelliJ AsciiDoc EAP repository](https://github.com/asciidoctor/asciidoctor-intellij-plugin/blob/master/CONTRIBUTING.adoc#eap).
